### PR TITLE
[Core] Checking number of nodes before operate in `ResidualBasedBossakDisplacementScheme`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This fails in MPI if the partition is empty.

**🆕 Changelog**

- [Checking number of nodes before operate in `ResidualBasedBossakDisplacementScheme`](https://github.com/KratosMultiphysics/Kratos/commit/65508f24b69a43c284be54d3932766ad0dc3ad0b)
- Minor clean up.

### CODE REVIEW

**Comment by roigcarlo** [comment]
Where is the change @loumalouomega? Its imposible to read with so many format changes....

File: kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp [hunk 1]
```diff
@@ -221,73 +237,75 @@ class ResidualBasedBossakDisplacementScheme
         const double delta_time = r_current_process_info[DELTA_TIME];
 
         // Updating time derivatives (nodally for efficiency)
-        const auto it_node_begin = rModelPart.Nodes().begin();
+        if (rModelPart.Nodes().size() > 0) {
```
Comment (loumalouomega) [inline kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp:240]: This is the actual change (rest is tabulation, and previous one is typedef ->using)

File: kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp [hunk 2]
```diff
@@ -325,42 +343,44 @@ class ResidualBasedBossakDisplacementScheme
         mBossak.c5 = ( delta_time * 0.5 * ( ( mBossak.gamma / mBossak.beta ) - 2.0 ) );
 
         // Updating time derivatives (nodally for efficiency)
-        const auto it_node_begin = rModelPart.Nodes().begin();
-
-        // Getting dimension
-        KRATOS_WARNING_IF("ResidualBasedBossakDisplacementScheme", !r_current_process_info.Has(DOMAIN_SIZE)) << "DOMAIN_SIZE not defined. Please define DOMAIN_SIZE. 3D case will be assumed" << std::endl;
-        const std::size_t dimension = r_current_process_info.Has(DOMAIN_SIZE) ? r_current_process_info.GetValue(DOMAIN_SIZE) : 3;
+        if (rModelPart.Nodes().size() > 0) {
```
Comment (loumalouomega) [inline kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp:346]: Idem (checking number of nodes)

**Review by roigcarlo — APPROVED** [review]

**Comment by loumalouomega** [comment]
<3

**Pull request merged**